lib/rspec/matchers/dsl.rb

```diff
@@ -521,11 +521,16 @@ def actual_arg_for(block)
         # rspec-rails so that it can define matchers that wrap
         # Rails' test helper methods, but it's also a useful
         # feature in its own right.
-        def method_missing(method, *args, &block)
-          if @matcher_execution_context.respond_to?(method)
-            @matcher_execution_context.__send__ method, *args, &block
-          else
-            super(method, *args, &block)
+        if RUBY_VERSION.to_f >= 2.7
```

Comment:
Any keyword argument supporting ruby supports this, so we can use the `RubyFeatures.kw_args_supported?` feature detection instead.
```suggestion
        if RubyFeatures.kw_args_supported?
```